lib/interfaces/tdd.js

```diff
@@ -84,6 +86,18 @@ module.exports = function (suite) {
       if (suite.isPending()) {
         fn = null;
       }
+
+      // Check for nested test registration
+      if (global._mochaExecutionState && global._mochaExecutionState.currentRunnable) {
```

Comment:
> `global.`

😬 We really don't want to add to the global scope if at all possible. Globals have a bad habit of getting mucked around with by other tools and exposing internal state.  Even when they're prefixed with a tool-specific name like `_mocha`. Plus they become very tricky to work with in concurrent/parallel scenarios.

Have you tried adding to the current test context? Searching on `context` and `currentContext` in `lib/` shows a few places that that's used to set up state. I would think populating a property like `currentContext.currentTest` would be a good path here.

In fact, searching for `currentTest` shows that it already exists! So something like `common`, `context`, or `currentTest` should be a place you can put things like this.

WDYT?